compiler/src/codegen/compcore.re

```diff
@@ -2889,52 +3941,42 @@ let compile_wasm_module = (~env=?, ~name=?, prog) => {
     | Some(e) => e
     };
   let (env, prog) = prepare(env, prog);
-  let funcs = compile_functions(env, prog);
-  let imports = compile_imports(env, prog);
-  let exports = compile_exports(env, prog);
-  let globals = compile_globals(env, prog);
-  let tables = compile_tables(env, prog);
-  let elems = compile_elems(env, prog);
-  let types = List.map(add_dummy_loc, BatDeque.to_list(env.func_types^));
-  let ret =
-    add_dummy_loc({
-      ...empty_module,
-      funcs,
-      imports,
-      exports,
-      globals,
-      tables,
-      elems,
-      types,
-      start: None,
-    });
-  validate_module(~name?, ret);
-  ret;
+  let wasm_mod = Module.create();
+  let _ = Module.set_features(wasm_mod, [Features.mvp, Features.multivalue]);
+  let _ = Memory.set_memory(wasm_mod, 0, max_int, "memory", [], false);
+  let () = ignore @@ compile_functions(wasm_mod, env, prog);
+  let () = ignore @@ compile_imports(wasm_mod, env, prog);
+  let () = ignore @@ compile_exports(wasm_mod, env, prog);
+  let () = ignore @@ compile_globals(wasm_mod, env, prog);
+  let () = ignore @@ compile_tables(wasm_mod, env, prog);
+  let () = ignore @@ compile_elems(wasm_mod, env, prog);
+  
+  validate_module(~name?, wasm_mod);
+  // TODO: Enable Binaryen optimizations
+  // Module.optimize(wasm_mod);
```

Comment:
Story behind this? Should be uncommented or better documented why it remains commented

Reply:
Doesn't work and needs investigation as to why, but it's probably a problem with the bindings. We have complete parity without the optimizations (we didn't optimize our WASM output before), but we will want to turn this on, so I left it with a TODO. I planned to open an issue for it after this was merged, but I can do it beforehand.